sdk/storage/storage-file/src/DirectoryClient.ts

```diff
@@ -348,25 +352,37 @@ export class DirectoryClient extends StorageClient {
   public async create(
     options: DirectoryCreateOptions = {}
   ): Promise<Models.DirectoryCreateResponse> {
-    if (!options.fileAttributes) {
-      options = validateAndSetDefaultsForFileAndDirectoryCreateCommonOptions(options);
-      // By default set it as a directory.
-      const attributes: FileSystemAttributes = new FileSystemAttributes();
-      attributes.directory = true;
-      options.fileAttributes = attributes;
-    }
-
-    return this.context.create(
-      fileAttributesToString(options.fileAttributes!),
-      fileCreationTimeToString(options.creationTime!),
-      fileLastWriteTimeToString(options.lastWriteTime!),
-      {
-        abortSignal: options.abortSignal,
-        metadata: options.metadata,
-        filePermission: options.filePermission,
-        filePermissionKey: options.filePermissionKey
+    const { span, spanOptions } = createSpan("DirectoryClient-create", options.spanOptions);
```

Comment:
I am missing C# `nameof` feature already...

Reply:
That would be so nice!